src/dataprocessing-mcp-server/awslabs/dataprocessing_mcp_server/utils/aws_helper.py

```diff
@@ -0,0 +1,167 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""AWS helper for the DataProcessing MCP Server."""
+
+import boto3
+import os
+from .consts import (
+    DEFAULT_RESOURCE_TAGS,
+    MCP_CREATION_TIME_TAG_KEY,
+    MCP_MANAGED_TAG_KEY,
+    MCP_MANAGED_TAG_VALUE,
+    MCP_RESOURCE_TYPE_TAG_KEY,
+)
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+
+class AwsHelper:
+    """Helper class for AWS operations.
+
+    This class provides utility methods for interacting with AWS services,
+    including region and profile management and client creation.
+    """
+
+    @staticmethod
+    def get_aws_region() -> Optional[str]:
+        """Get the AWS region from the environment if set."""
+        return os.environ.get(
+            'AWS_REGION',
+        )
+
+    @staticmethod
+    def get_aws_profile() -> Optional[str]:
+        """Get the AWS profile from the environment if set."""
+        return os.environ.get('AWS_PROFILE')
+
+    # Class variable to cache the AWS account ID
+    _aws_account_id = None
+
+    @classmethod
+    def get_aws_account_id(cls) -> str:
+        """Get the AWS account ID for the current session.
+
+        The account ID is cached after the first call to avoid repeated STS calls.
+
+        Returns:
+            The AWS account ID as a string
+        """
+        # Return cached account ID if available
+        if cls._aws_account_id is not None:
+            return cls._aws_account_id
+
+        try:
+            sts_client = boto3.client('sts')
+            cls._aws_account_id = sts_client.get_caller_identity()['Account']
+            return cls._aws_account_id
+        except Exception:
+            # If we can't get the account ID, return a placeholder
+            # This is better than nothing for ARN construction
+            return 'current-account'
+
+    @classmethod
+    def create_boto3_client(cls, service_name: str, region_name: Optional[str] = None) -> Any:
+        """Create a boto3 client with the appropriate profile and region.
+
+        The client is configured with a custom user agent suffix 'awslabs/mcp/dataprocessing-mcp-server/0.1.0'
+        to identify API calls made by the Dataprocessing MCP Server.
+
+        Args:
+            service_name: The AWS service name (e.g., 'ec2', 's3', 'glue', 'emr-ec2')
+            region_name: Optional region name override
+
+        Returns:
+            A boto3 client for the specified service
+        """
+        # Get region from parameter or environment if set
+        region: Optional[str] = region_name if region_name is not None else cls.get_aws_region()
+
+        # Get profile from environment if set
+        profile = cls.get_aws_profile()
+
+        # Create config with user agent suffix
+        config = Config(user_agent_extra='awslabs/mcp/dataprocessing-mcp-server/0.1.0')
```

Comment:
would need to add `from awslabs. dataprocessing_mcp_server import __version__`
```suggestion
        config = Config(user_agent_extra='awslabs/mcp/dataprocessing-mcp-server/{__version__}')
```